ratis-client/src/main/java/org/apache/ratis/client/impl/OrderedStreamAsync.java

```diff
@@ -147,4 +156,13 @@ private void sendRequestToNetwork(DataStreamWindowRequest request){
       return null;
     });
   }
+
+  private void scheduleWithTimeout(DataStreamWindowRequest request, TimeDuration sleepTime) {
+    scheduler.onTimeout(sleepTime, () -> {
+      if (!request.getReplyFuture().isDone()) {
+        request.getReplyFuture().completeExceptionally(
+            new IOException("Send request timeout " + request));
```

Comment:
- Let's use TimeoutIOException instead of IOException
- Include the timeout value in the error message of TimeoutIOException.
```
            new TimeoutIOException("Timeout " + requestTimeout + ": Failed to send " + request));
```